### DISCUSSION TITLE
Argos produces 0 byte "string_map.dat" files

### DISCUSSION THREAD

**Discussion by quasiquote1** [sparcians/map#420] [Q&A] [answered]
Hi,

I am trying to enable Argos to dump based on cycle ranges (using the built-in "--debug-on" and "-r" command-line parameters).

However I found that this leads to empty string_map.dat files, which argos.py then emits a runtime error about when viewing the file.  There are no errors during the runtime production with this file however.

I tried dumping from cycle 1 and this works (and produces non-empty string_map.dat files).

Is this a known bug, and is there a workaround for this issue?

Thanks!

**Comment by klingaard** [comment]
Hmmm... dumb question, are you doing this?

```
--debug-on 100k -r 10k
```
expecting the simulator to turn debugging on after 100K cycles and run an additional 10k?

**Reply by quasiquote1** [reply]
No, I did something like 

```
--debug-on 100k -r 101k

**Reply by klingaard** [reply]
Does your simulator output a message indicating that it started collection?  What happens if you go another 1k?

**Reply by klingaard** [reply]
Curious, have you had a chance to try my suggestions?

**Reply by quasiquote1** [reply, marked as answer]
Thanks for the suggestion - I was able to have non-empty string_map.dat if I increased to 106k cycles.  It might be good to give a warning about this file is empty during generation time so the user is aware of this.

**Reply by klingaard** [reply]
That's a good idea.  Both in argos and on generation.  Interesting it took that long to see data.
